neqo-crypto/src/agent.rs

```diff
@@ -572,6 +708,30 @@ impl SecretAgent {
         })
     }
 
+    /// Install a certificate compression mechanism.
+    ///
+    /// # Errors
+    /// If the compression mechanism with the same id is already registered
+    /// If too many compression mechanisms are already registered
+    ///
+    /// This returns an error if the certificate compression could not be established
+    ///
+    /// [RFC8879]: https://datatracker.ietf.org/doc/rfc8879/
+    pub fn set_certificate_compression<T: CertificateCompressor>(&mut self) -> Res<()> {
+        if T::ID == 0 {
+            return Err(Error::InvalidCertificateCompressionID);
+        }
```

Comment:
I like this, but you could make this structural if you change the type of ID to `NonZeroU16`.

It's a bit more annoying to use:

```rust
   const ID: NonZeroU16 = NonZeroU16::new(12).unwrap();
```

But you save on the need for a new error enum member.

Reply:
@Frosne Wanted to check if you are planning to change this, or whether I should merge as-is?